### PR TITLE
Add support for stashing the CLI program names in a common way

### PR DESCRIPTION
**Pull request by tristan957**
Many of our tools store the name of the program, typically for use in usage statements. All of the programs that store the name did so in custom ways and created repeated code.

The pattern we use fairly often, but not always, in CLIs is output taking the form of `<progname>: <message>`. By storing the program name in libhse-cli, we can more effectively move to a paradigm where we don't create multiple `fatal()`, `syntax()`, etc. functions per program, though this improvement can come at a later time.

Signed-off-by: Tristan Partin <tpartin@micron.com>

### CODE REVIEW

**Comment by tristan957** [comment]
Went down the road of fixing the nightly tests, ended up on this tangent for the solution that I want to take for solving the test failures.

File: lib/binding/kvdb_interface.c [hunk 1]
```diff
@@ -411,7 +409,11 @@ hse_init(const char *const config, const size_t paramc, const char *const *const
     }
 
     /* First log message w/ HSE version - after deserializing global params */
-    log_info("version %s, program %s", HSE_VERSION_STRING, hse_progname);
+#ifdef _GNU_SOURCE
+    log_info("version %s, program %s", HSE_VERSION_STRING, program_invocation_short_name);
```
Comment (tristan957) [inline lib/binding/kvdb_interface.c:413]: One could argue how useful it is to log the program name when syslog already takes care of this if you don't call openlog() with a custom identifier.

Just a thought.

File: lib/cn/cn_metrics.h [hunk 2]
```diff
@@ -6,6 +6,7 @@
 #ifndef HSE_KVDB_CN_METRICS_H
 #define HSE_KVDB_CN_METRICS_H
 
+#include <assert.h>
```
Comment (tristan957) [inline lib/cn/cn_metrics.h:9]: Some combination of headers churn caused this to be necessary for the one assert statement in this header.

File: tools/ptree-overload/ptree-overload.c [hunk 3]
```diff
@@ -93,8 +91,8 @@ usage(void)
         "-k nkeys   Number of keys\n"
         "-p nptombs Number of ptombs\n"
         "-Z config  Path to global config file\n"
-        "-h      Print this help menu\n"
-        , progname);
+        "-h         Print this help menu\n",
+        progname);
```
Comment (beckerg) [inline tools/ptree-overload/ptree-overload.c:95]: Myabe as well alphabetize the options while you're here, AaBbCcDd...

File: lib/binding/kvdb_interface.c (hunk 1 above)
Comment (beckerg) [reply lib/binding/kvdb_interface.c:413]: The original intent was to print the full path, so maybe change this to "program_invocation_name".

**Review by beckerg — APPROVED** [review]

File: lib/binding/kvdb_interface.c (hunk 1 above)
Comment (tristan957) [reply lib/binding/kvdb_interface.c:413]: Ok, I will do that.

File: tools/ptree-overload/ptree-overload.c (hunk 3 above)
Comment (tristan957) [reply tools/ptree-overload/ptree-overload.c:95]: Sure will do.

**Pull request merged**